### PR TITLE
enhanced documentation

### LINKED ISSUE #470: Include the missing dependencies in the documentation
- [ ] Find and include the missing dependencies on the documentation of the "installation instructions"

- [ ] Improve or fix the tutorial in the documentation

### PR DESCRIPTION
**Pull request by myrtosh**
closes #470

### CODE REVIEW

File: doc/content/introduction/installation/_index.md [hunk 1]
```diff
@@ -136,7 +144,3 @@ processes are used that deploy build jobs to a number of build
 clients. In most scenarios, the master images need to be built on the
 machine where the software build is started.
 
-[^gopath_deprecation]: Older versions of the documentation contained references to $GOPATH
-    and recommendations to install programs into $GOPATH/bin. Since QMSTR 0.4,
-    GOPATH is not used anymore. The Makefiles instead expect that program
-    dependencies like _protoc_ to be found in the PATH.
```
Comment (senaingrid) [inline doc/content/introduction/installation/_index.md:142]: I think we should keep it. What do you think @GiasemiSh ?

File: masterserver/Dockerfile [hunk 2]
```diff
@@ -9,8 +9,10 @@ ENV DEBIAN_FRONTEND=noninteractive
 
 # install runtime deps
 RUN apt-get update && apt-get install -y \
-    software-properties-common curl autoconf git apt-utils libgit2-dev libio-captureoutput-perl python python-pip python3-distutils protobuf-compiler icu-devtools libicu-dev && \
+    software-properties-common curl autoconf git apt-utils libgit2-dev libio-captureoutput-perl unzip python python-pip python3-distutils icu-devtools libicu-dev && \
     rm -rf /var/lib/apt/lists/*
+RUN curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v3.11.4/protoc-3.11.4-linux-x86_64.zip
```
Comment (GiasemiSh) [inline masterserver/Dockerfile:14]: Why do we need this? Please explain in your commit message why not to keep the previous installation.

File: doc/content/introduction/installation/_index.md [hunk 3]
```diff
@@ -65,22 +73,22 @@ Add user to the docker group:
...
 ## Installing the clients
 
 The main entry point into the installation tasks for Quartermaster is
 the Makefile in the main repository. The default installation prefix is `/usr/local`.
 The client programs will be installed into the `bin/` subdirectory:
   
+  > cd qmstr/
+
+  > go install github.com/golang/protobuf/protoc-gen-go
+
```
Comment (GiasemiSh) [inline doc/content/introduction/installation/_index.md:91]: This should be included in the section ##Prerequisites, it's not related here.

File: doc/content/introduction/installation/_index.md [hunk 4]
```diff
@@ -65,22 +73,22 @@ Add user to the docker group:
...
+  
 
 ## Installing the clients
 
 The main entry point into the installation tasks for Quartermaster is
 the Makefile in the main repository. The default installation prefix is `/usr/local`.
 The client programs will be installed into the `bin/` subdirectory:
   
+  > cd qmstr/
+
```
Comment (GiasemiSh) [inline doc/content/introduction/installation/_index.md:89]: This is not needed

File: doc/content/introduction/installation/_index.md (hunk 1 above)
Comment (GiasemiSh) [reply doc/content/introduction/installation/_index.md:142]: Yes true, we should keep this comment.

Comment (zetoph) [reply doc/content/introduction/installation/_index.md:142]: One question is why do we have the GOPATH as variable in our Makefile if we don't need it? Probably should check if it's safe to remove. If not, this comment is misleading.

File: doc/content/introduction/installation/_index.md (hunk 3 above)
Comment (myrtosh) [reply doc/content/introduction/installation/_index.md:91]: To install protoc-gen-go you need to be in the same directory as go.mod thus inside qmstr.

File: doc/content/introduction/installation/_index.md (hunk 4 above)
Comment (myrtosh) [reply doc/content/introduction/installation/_index.md:89]: I used it to make sure that the user installs protoc-gen-go inside the qmstr directory

Comment (myrtosh) [reply doc/content/introduction/installation/_index.md:89]: otherwise it fails

File: doc/content/introduction/installation/_index.md [hunk 5]
```diff
@@ -144,3 +144,8 @@ processes are used that deploy build jobs to a number of build
 clients. In most scenarios, the master images need to be built on the
 machine where the software build is started.
 
+
+Older versions of the documentation contained references to $GOPATH
+and recommendations to install programs into $GOPATH/bin. Since QMSTR 0.4,
+GOPATH is not used anymore. The Makefiles instead expect that program
+dependencies like _protoc_ to be found in the PATH.
```
Comment (GiasemiSh) [inline doc/content/introduction/installation/_index.md:151]: Better to add this as a note and not a normal paragraph in the `summary` section.

**Comment by fullsushidev** [comment]
Closing this PR since we changed the way we're going install QMSTR from now own.

**Pull request closed without merging**